packages/vite/src/node/server/middlewares/proxy.ts

```diff
@@ -14,6 +14,10 @@ export interface ProxyOptions extends HttpProxy.ServerOptions {
    * rewrite path
    */
   rewrite?: (path: string) => string
+  /**
+   * dynamically set the target
+   */
+  router?: (req: http.IncomingMessage) => HttpProxy.ServerOptions['target']
```

Comment:
Instead of adding a new property, would you change the existing `target` property to receive a function?